### PR TITLE
Compute mean IC50

### PR DESCRIPTION
**Pull request by pva701**
## Related issue(s)

https://issues.serokell.io/issue/EDNA-119

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [ ] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [ ] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [ ] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: frontend/src/store/selectors.ts [hunk 1]
```diff
@@ -175,25 +176,39 @@ export const selectedSubExperimentsQuery = selector<SubExperimentWithMeasurement
...
-        "Left" in acc
-          ? acc
-          : "Left" in x.meta.item.result
-          ? { Left: x.meta.item.result.Left }
-          : { Right: acc.Right + x.meta.item.result.Right[2] },
-      { Right: 0 }
-    );
-    if ("Left" in ic50Sum) return ic50Sum;
-    return { Right: ic50Sum.Right / subs.length };
+    return evalMeanIC50(get(selectedSubExperimentsMetaQuery).map(x => x.item.result));
```
Comment (gromakovsky) [inline frontend/src/store/selectors.ts]: The issue says:
> Somewhere in Dashboard we show mean IC50 for all primary sub-experiments of all experiments where at least one sub-experiment is selected. I think that's what they want, not mean for all selected sub-experiments.

Currently we calculate average for all sub-experiments shown on plot, but I think they want each selected experiment to have equal weight, even if multiple sub-experiments are selected. They say:
> Необходим расчет среднего значения IC50 при выборе нескольких экспериментов

It's formulated imprecisely, so maybe I am wrong. Maybe it's better to have the following logic: if multiple sub-experiments are chosen from one experiment, we calculate average for them and then we take average of all these "averages". So if you have `se1` and `se2` from one experiment and `se3` for another experiment, we can return `((se1 + se2) / 2 + se3) / 2`. I don't insist on doing that and I don't know what they want. If you think I am saying nonsense, feel free to ignore. In general please try to maximize usefulness of this value divided by complexity of implementing it.

Comment (pva701) [reply frontend/src/store/selectors.ts]: >So if you have se1 and se2 from one experiment and se3 for another experiment, we can return ((se1 + se2) / 2 + se3) / 2

Ok, maybe it makes sense... it's not hard to implement

File: frontend/src/components/IC50Line/IC50Tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import { Loadable } from "recoil";
+import React from "react";
+import { ResultDto } from "../../api/types";
+import { Tooltip } from "../Tooltip/Tooltip";
+import { formatIC50 } from "../../utils/utils";
+import "./IC50Tooltip.scss";
+
+export type IC50Value = ResultDto<number | number[]> | Loadable<ResultDto<number | number[]>>;
+
+interface IC50LineProps {
```
Comment (sashasashasasha151) [inline frontend/src/components/IC50Line/IC50Tooltip.tsx]: Tooltip props

File: frontend/src/components/IC50Line/IC50Tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  ic50: IC50Value;
+  className?: string;
+}
+
+export function IC50Tooltip({ ic50, className }: IC50LineProps): React.ReactElement {
```
Comment (sashasashasasha151) [inline frontend/src/components/IC50Line/IC50Tooltip.tsx]: You write Tooltip here 4 times, but they are same all time. Like you can just return from some function all information which you need for Tooltip and write only one Tooltip

**Review by sashasashasasha151 — APPROVED** [review]

File: frontend/src/components/IC50Line/IC50Tooltip.tsx (hunk 3 above)
Comment (pva701) [reply frontend/src/components/IC50Line/IC50Tooltip.tsx]: nichego ne ponyal

**Review by gromakovsky — APPROVED** [review]

**Pull request merged**